### DISCUSSION TITLE
How to code for secondary task response time?

### DISCUSSION THREAD

**Discussion by zoexxu** [jspsych/jsPsych#1423] [Q&A]
Hi there!

I learned about jsPsych through the training videos uploaded on YouTube. Thanks for the generosity! They are very useful.
I need help with setting up Secondary Task Response Time study with javaScript. I wonder if anyone can help me with it. Basically the study runs as follows:
Participants will watch videos as the primary task, during which visual signals (e.g., an image of a question mark) will be presented on the screen. Upon noticing the visual signals, participants are expected to press a button (e.g., space bar) as quickly as possible. The pressing button practice is the secondary task. I'm interested in the response time to the secondary task. 
Supposing participants will see three videos. During each video, an image of a question mark as visual signal will appear three times randomly both in location on the screen and time. The videos will not be disrupted (e.g., paused) due to the presence of the visual signals. 
How to code with javaScript for this study design? Is there any packages available in the library to solve this problem? I'm new to javaScript, so any suggestions are appreciated!

**Comment by jodeleeuw** [comment]
Hi @zoexxu,

You'll need to create a new plugin to accomplish this. You could start with the `video-keyboard-response` plugin as the foundation, and add in the two additional features (showing the secondary task stimulus, measuring response time to it). 

It's a somewhat complicated set of features to add, so I can't offer you the full solution here, but I can provide some pointers.

For showing the secondary stimulus, you'll probably want to use the method `display_element.insertAdjacentHTML()` to add the secondary task to the screen. You'll need to use CSS rules to control the position of this element. I'd suggest something like `position: fixed; top: 20%; left: 30%;` for example. You can find lots of documentation about these methods online. The [Mozilla Developers Network](https://developer.mozilla.org/en-US/) is an authoritative source for JavaScript reference.

To record the RTs to the secondary stimulus, you'll need to use `jsPsych.pluginAPI.getKeyboardResponse()` after showing the stimulus on the screen. You can see examples of how to use this method in the various `*-keyboard-response` plugins, and of course the jsPsych docs.

Good luck with the plugin, and if you have specific questions that you run into while implementing it feel free to post those here!

**Reply by zoexxu** [reply]
Thank you very much! I'll give it a try :)

**Comment by becky-gilbert** [comment]
Hi @zoexxu, a colleague of mine has a jsPsych plugin that does this. It's probably not exactly what you need, since it's customized for their experiment, but might still be helpful as a reference. I'll see if I can share it and post here with an update.

**Reply by zoexxu** [reply]
Sounds great! Thank you!
